Forta-Bots-Challenge-2/src/agent.ts

```diff
@@ -0,0 +1,35 @@
+import { Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+import { UNISWAP_V3_FACTORY_ADDRESS, SWAP_EVENT } from "./constants";
+import FactoryFetcher from "./factory.fetcher";
+import { providers } from "ethers";
+import { createFinding } from "./findings";
+import { Interface } from "@ethersproject/abi";
+
+export const provideHandleTransaction = (
+  swapEvent: string,
+  uniswapFactoryAddress: string,
+  provider: providers.Provider
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+    const fetcher: FactoryFetcher = new FactoryFetcher(provider);
```

Comment:
Typically, when we use classes like this `FactoryFetcher` like you did, we might create it in an `initialize()` function from the SDK. That is so we wouldn't need to keep creating class instances in _evey_ transaction, which is what would happen here.

With the correct usage of the `initialize()` function, you would only create it once, making your bot more efficient. However, really think if that is necessary.

Great to see you attempt to use one and creating a test suite for it though!